rapthor/lib/sector.py

```diff
@@ -304,11 +324,12 @@ def make_skymodel(self, index):
             self.central_patch = patch_names[patch_dist.index(min(patch_dist))]
 
             # Filter the field source sky model and store source sizes
-            all_source_names = self.field.source_skymodel.getColValues('Name').tolist()
-            source_names = skymodel.getColValues('Name')
-            in_sector = np.array([all_source_names.index(sn) for sn in source_names])
-            source_skymodel = self.field.source_skymodel.copy()
-            source_skymodel.select(in_sector)
+            if source_skymodel is None:
```

Comment:
I'm a bit confused by the use of `source_skymodel`. It seems to act as some kind of boolean, where `None` means `False`, and `[]` means `True`. Am I overlooking something here?

Reply:
Yeah, it is confusing. I would like to refactor all this at some point. Basically, `None` means it still needs to be evaluated, and `[]` just means it has zero length (for the `len()` check that appears later).